src/lib/token_list_gen.ts

```diff
@@ -285,7 +289,31 @@ export const generateTokenList = async (
     if (uri) logoUris[token.l1TokenAddr] = uri;
   }
 
-  let arbifiedTokenList: ArbTokenInfo[] = tokens
+  const getL1GatewayAddress = async (l2GatewayAddress: string) => {
+    let l2Gateway: string;
+    if (isNova) {
+      if (
+        (l2Gateway = l2ToL1GatewayAddressesNova[l2GatewayAddress.toLowerCase()])
+      ) {
+        return l2Gateway;
+      }
+    } else {
+      if ((l2Gateway = l2ToL1GatewayAddresses[l2GatewayAddress.toLowerCase()]))
+        return l2Gateway;
+    }
```

Comment:
```suggestion
    let l2Gateway: string = isNova
      ? l2ToL1GatewayAddressesNova[l2GatewayAddress.toLowerCase()]
      : l2ToL1GatewayAddresses[l2GatewayAddress.toLowerCase()];
    if (l2Gateway) {
      return l2Gateway;
    }
```
Personally I'd prefer this for better readability 🧐
or add a comment for future readers
but I'll defer it to others